client/compliance_client.go

```diff
@@ -141,51 +162,54 @@ func NewComplianceClient(ctx context.Context, opts ...option.ClientOption) (*Com
 	if err != nil {
 		return nil, err
 	}
-	c := &ComplianceClient{
+	client := ComplianceClient{CallOptions: defaultComplianceCallOptions()}
+
+	c := &complianceGRPCClient{
 		connPool:         connPool,
 		disableDeadlines: disableDeadlines,
-		CallOptions:      defaultComplianceCallOptions(),
-
 		complianceClient: genprotopb.NewComplianceClient(connPool),
+		CallOptions:      &client.CallOptions,
 	}
 	c.setGoogleClientInfo()
 
+	client.internalComplianceClient = c
+
 	c.operationsClient = longrunningpb.NewOperationsClient(connPool)
 
 	c.iamPolicyClient = iampb.NewIAMPolicyClient(connPool)
 
 	c.locationsClient = locationpb.NewLocationsClient(connPool)
 
-	return c, nil
+	return &client, nil
```

Comment:
nit: for the three assignments above: `c.fooClient = ...`: does it make sense to put them inside the literal `c := &complianceGRPCClient{...}` definition for brevity/clarity?

Reply:
Setting them separately was the convention in the existing code, but I agree, it's cleaner.